pkg/monitoring/vmistats/collector.go

```diff
@@ -250,7 +250,7 @@ func (co *VMICollector) updateVMIsPhase(vmis []*k6tv1.VirtualMachineInstance, ch
 		mv, err := prometheus.NewConstMetric(
 			vmiCountDesc, prometheus.GaugeValue,
 			float64(count),
-			vmc.NodeName, vmc.Phase, vmc.OS, vmc.Workload, vmc.Flavor, vmc.InstanceType,
+			vmc.NodeName, vmc.Phase, vmc.OS, vmc.Workload, vmc.Flavor, vmc.InstanceType, vmc.Preference,
```

Comment:
Can you please elaborate on why the absence of `vmc.Preference` caused the metric to not be pushed and not to be collected?

Reply:
`NewConstMetric` was throwing an error saying the number of labels set in `vmiCountDesc` and the number of labels used as arguments differed. The issue was being logged.

Also, the unit tests were incomplete, testing only the reading of the VMIs details, and were missing the creation of the metric